src/Service/HelloRetailClientService.php

```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+
+namespace Helret\HelloRetail\Service;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Request;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+
+class HelloRetailClientService
+{
+    private const url = "https://core.helloretail.com/serve/";
+    private const userEndpoint = "trackingUser";
+    private string|null $apiKey = null;
+    private Client $client;
+
+    public function __construct(protected SystemConfigService $systemConfigService)
+    {
+        $this->client = new Client();
```

Comment:
@wexoigi is is not ideal to instantiate new classes and load data in the constructor, since the constructor will be called A LOT even when not used.
Better to move client instantiation to the loadAuthData function and then call loadAuthData function in each API call function